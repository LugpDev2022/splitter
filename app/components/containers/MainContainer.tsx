interface MainContainerProps {
  children: React.ReactNode;
}

const MainContainer: React.FC<MainContainerProps> = ({ children }) => {
  return (
    <div
      className=' 
      bg-white
      gap-7
      grid
      grid-cols-1
      lg:w-9/12
      md:grid-cols-2
      md:rounded-[30px]
      md:w-11/12
      p-7
      rounded-t-[35px]
      w-full
      '
    >
      {children}
    </div>
  );
};

export default MainContainer;
