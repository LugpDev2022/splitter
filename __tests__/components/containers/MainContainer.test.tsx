import MainContainer from '@/app/components/containers/MainContainer';
import { render, screen } from '@testing-library/react';

describe('Tests on <GeneralContainer />', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <MainContainer>
        <h2>Example h2</h2>
      </MainContainer>
    );

    expect(container).toMatchSnapshot();
  });

  test('should render the given content', () => {
    const testText = 'Example h2';

    render(
      <MainContainer>
        <h2>{testText}</h2>
      </MainContainer>
    );

    expect(screen.getByText(testText)).toBeTruthy();
  });
});
