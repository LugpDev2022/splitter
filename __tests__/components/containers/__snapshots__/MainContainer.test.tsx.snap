// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GeneralContainer /> should match snapshot 1`] = `
<div>
  <div
    class="  bg-white gap-7 grid grid-cols-1 lg:w-9/12 md:grid-cols-2 md:rounded-[30px] md:w-11/12 p-7 rounded-t-[35px] w-full "
  >
    <h2>
      Example h2
    </h2>
  </div>
</div>
`;
